tests/pytests/functional/utils/test_process.py

```diff
@@ -35,3 +39,36 @@ def test_process_manager_60749(process_manager):
 
     process_manager.add_process(Process)
     process_manager.check_children()
+
+
+def _get_num_fds(pid):
+    "Determine the number of open fds for a process, linux only."
+    proc = subprocess.run(
+        ["ls", "-l", f"/proc/{pid}/fd"], capture_output=True, check=True
+    )
+    return len(proc.stdout.decode().split("\n")) - 1
```

Comment:
I recommend just using python functions like `os.listdir()` or Path().iterdir()` to list the files in the directory instead of forking a subprocess.

Reply:
Good call.